lib/round-tracker.js

```diff
@@ -30,8 +30,18 @@ export const MAX_TASKS_PER_NODE = 15
 export async function startRoundTracker ({ pgPool, signal }) {
   const contract = await createMeridianContract()
 
-  const onRoundStart = (newRoundIndex) => {
-    updateSparkRound(pgPool, contract, newRoundIndex).catch(err => {
+  const onRoundStart = (newRoundIndex, ...args) => {
+    /** @type {import('ethers').ContractEventPayload} */
+    const event = args.pop()
+    const blockNumber = event?.log?.blockNumber
+    if (blockNumber === undefined) {
+      console.warn(
+        'Ethers.js event data did not include ContractEventPayload with the block number. ' +
+        'Will call `eth_getLogs` to find the round start epoch.'
+      )
+    }
```

Comment:
This seems to work very well. 👍🏻 We are advancing rounds, and the warning is not printed.

```
2024-05-21T08:56:31Z app[5683210c621628] cdg [info]Mapped 0x8460766Edc62B525fc1FA4D628FC79229dC73031 IE round index 7310n to SPARK round number 12733n (start epoch: 3932512)
2024-05-21T08:56:33Z app[5683210c621628] cdg [info]SPARK round started: 12733n (epoch: 3932512)
2024-05-21T08:56:33Z app[9185e760f41658] cdg [info]Mapped 0x8460766Edc62B525fc1FA4D628FC79229dC73031 IE round index 7310n to SPARK round number 12733n (start epoch: 3932512)
2024-05-21T08:56:33Z app[9185e760f41658] cdg [info]SPARK round started: 12733n (epoch: 3932512)
2024-05-21T08:56:33Z app[3d8dd965ae2998] cdg [info]Mapped 0x8460766Edc62B525fc1FA4D628FC79229dC73031 IE round index 7310n to SPARK round number 12733n (start epoch: 3932512)
2024-05-21T08:56:33Z app[3d8dd965ae2998] cdg [info]SPARK round started: 12733n (epoch: 3932512)
2024-05-21T08:56:34Z app[48ed67dc0e0308] cdg [info]Mapped 0x8460766Edc62B525fc1FA4D628FC79229dC73031 IE round index 7310n to SPARK round number 12733n (start epoch: 3932512)
2024-05-21T08:56:34Z app[48ed67dc0e0308] cdg [info]SPARK round started: 12733n (epoch: 3932512)
```